### PR TITLE
chore: Add id and state to CaseData

### PR DESCRIPTION
**Pull request by tomaszpowroznik**
- add case id to CaseData
- add state to CaseData
- add CaseConverter to convert CaseDetials to CaseData
limit usage of CaseDetails and CallbackRequest to Controllers only, services and event handlers operates on CaseData
- remove EventData
- abstract CallbackController to provide common utilities

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java [hunk 1]
```diff
@@ -642,4 +643,6 @@ public boolean hasFutureHearing(List<Element<HearingBooking>> hearingBookings) {
             .filter(hearingBooking -> hearingBooking.getStartDate().isAfter(currentCmoStartDate))
             .min(comparing(HearingBooking::getStartDate));
     }
+
+    private String sendToCtsc;
```
Comment (KiranVarma02) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java:647]: nice. I created a sub-task for this and other scenarios like this - 

https://tools.hmcts.net/jira/browse/FPLA-2148

File: service/src/main/java/uk/gov/hmcts/reform/fpl/events/CaseEvent.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.hmcts.reform.fpl.events;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+@Getter
+@EqualsAndHashCode
+public class CaseEvent {
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/events/CaseEvent.java:9]: I don't think that this has been used, are all events supposed to extend it?

Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/events/CaseEvent.java:9]: you are right this one is orphaned, will remove

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/CoreCaseDataStoreLoader.java [hunk 3]
```diff
@@ -13,10 +20,21 @@
 
 public class CoreCaseDataStoreLoader {
 
-    private static final ObjectMapper mapper = new ObjectMapper();
+    private static final ObjectMapper mapper = JsonMapper.builder()
```
Comment (akomisarek) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/utils/CoreCaseDataStoreLoader.java:23]: I wonder if we could just inject the ObjectMapper from spring.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/CoreCaseDataStoreLoader.java [hunk 4]
```diff
@@ -13,10 +20,21 @@
...
+        .addModule(new ParameterNamesModule())
+        .addModule(new Jdk8Module())
+        .addModule(new JavaTimeModule())
+        .build();
+
+    private static final CaseConverter caseConverter = new CaseConverter(mapper);
 
     private CoreCaseDataStoreLoader() {
-        // NO-OP
+        new ObjectMapper();        // NO-OP
```
Comment (akomisarek) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/utils/CoreCaseDataStoreLoader.java:33]: Really, does sonar complains? Even in tests? ;o 

What if there is no constructor, it's test class so I wouldn't worry..

**Review by akomisarek — APPROVED** [review]
I don't want to see this change anymore :P. 

Couple of comments, but feel free to change in follow up PR, and as we agreed the PopulateDirectionsHandler could use some refactoring as well to remove even more code.

**Pull request merged**